src/main/java/address/MainApp.java

```diff
@@ -35,15 +41,29 @@ public void start(Stage primaryStage) {
     }
 
     protected void setupComponents() {
+
         config = getConfig();
         PreferencesManager.setAppTitle(config.appTitle);
-        AddressBookWrapper dataFromFile = StorageManager.getDataFromFile(
-                                                PreferencesManager.getInstance().getPersonFile());
+
+        final File initialFile = PreferencesManager.getInstance().getPersonFile();
+        AddressBookWrapper dataFromFile;
+        Exception fileLoadException = null;
+        try {
+             dataFromFile = StorageManager.loadDataFromSaveFile(initialFile);
+        } catch (JAXBException | FileContainsDuplicatesException e) {
+            fileLoadException = e;
+            dataFromFile = new AddressBookWrapper();
+        }
+
```

Comment:
If we change `mainController`'s initialization to be independent of reading data from the file, then everything becomes a lot simpler. (We can use the normal load-from-save-file code without any problems)